### PR TITLE
Fix ecoBridge infinite loop

### LINKED ISSUE #1626: Bridge claiming function infinite loop
_Users are being sent back to the original chain from which a bridge transaction was executed when trying to execute a claim._

**Expected behaviour:** The claim function in the bridge tab sends the user to the correct chain and calls the claim function in the relevant bridge.

**Observed behaviour:** The claim function sends the user back to the initiated chain (possibly Eth mainnet?) and does nothing. This appears unrelated to the users' selected chain or bridge of choice, as reports have covered Ethereum mainnet, Gnosis Chain, Polygon, and Arbitrum. It also has encompassed several bridges from what I am able to tell at this time (Omnibridge, Connext, Arbitrum Native Bridge). 

**Collection of reports:**

image

image

image

image

_Seems like a polygon to Arb transfer was pushing the user to mainnet as well?_

**A possibly related report from 09/29/2022:**

image

_The user needed to visit the native bridge, although, stuck on pending, appears to be a separate, possibly solved issue compared to the above loop._

### PR DESCRIPTION
**Pull request by 0xVance**
# Summary

Fixes #1626

I think `_activeBridge` check was introduced to improve logging when error happends but it occured to be the error :bug:

### CODE REVIEW

**Comment by MilanVojnovic95** [comment]
Tested and:

USDC from Ethereum to Arbitrum (Socket) ✅ 
USDC from Ethereum to Gnosis (Connext) ✅ 
USDC from Arbitrum One to Gnosis (Socket) ✅ 

USDC from Gnosis to Polygon ❌ 

On Swapr it shows as if the bridge failed:
image


But on [socketscan](https://socketscan.io/tx/0x6744528858394805eab7f6896c1eb9aae8d413c50f9d80b617851f4ee9486968) it shows that it's still in progress:
image


USDC from Ethereum to Gnosis (OmniBridge) ❌ 

When user clicks on the "Collect" option, they receive the following error message:
image

**Comment by 0xVenky** [comment]
> image

Hey, this needs a fix for this PR. I think this is a different issue, but still, lets fix everything at once.

The other one related to Omnibridge, lets ignore that for now and create a seperate issue for this one and will try to fix it in the coming iterations.

**Comment by berteotti** [comment]
I made a transaction from Gnosis to Ethereum (Connext) and couldn't claim it on Ethereum. Don't know if this is fixed in this PR :)

**Comment by 0xVenky** [comment]
> I made a transaction from Gnosis to Ethereum (Connext) and couldn't claim it on Ethereum. Don't know if this is fixed in this PR :)

Hey, This PR should have fixed it. :)

**Review by Mi-Lan — APPROVED** [review]
- This fixes it! Although I get the same error message Milan outlined that gas estimation failed
Screenshot 2023-01-03 at 15 47 54
- We can tackle that in another issue though
- Im also getting infinite loading when waiting for confirmation for Omni bridge

**Review by adamazad — APPROVED** [review]

**Comment by 0xVance** [comment]
> Tested and:
> 
> USDC from Ethereum to Arbitrum (Socket) white_check_mark USDC from Ethereum to Gnosis (Connext) white_check_mark USDC from Arbitrum One to Gnosis (Socket) white_check_mark
> 
> USDC from Gnosis to Polygon x
> 
> On Swapr it shows as if the bridge failed: image
> 
> But on [socketscan](https://socketscan.io/tx/0x6744528858394805eab7f6896c1eb9aae8d413c50f9d80b617851f4ee9486968) it shows that it's still in progress: image
> 
> USDC from Ethereum to Gnosis (OmniBridge) x
> 
> When user clicks on the "Collect" option, they receive the following error message: image
> 
> Also another thing I noticed:
> 
> the status "failed" does not change, even when you go to check the transaction id status "bridge in progress" and even when "Bridge successful" on Swapr, it appears in the status "failed".

Imo Socket has a bug on /bridge-status endpoint.

The reason why status becomes 'Failed' is because /bridge-status endpoint returns 500 if called right after initiating bridging (which indicates some kind of bug).

When I bridged XDAI to USDC from Gnosis to Polygon I got this:
image
image

When I called this endpoint manually through their API it responded correctly.

My suggestion for fixing this:
- in pending tx listener for socket, when it responds with 500 don't change status to failed but just skip it for next 10s
- on app init go through all socket transactions with 'failed' status and retry them

I'll add a commit with this briefly

**Comment by 0xVance** [comment]
> * This fixes it! Although I get the same error message Milan outlined that gas estimation failed
> 
> Screenshot 2023-01-03 at 15 47 54
> 
> * We can tackle that in another issue though
> * Im also getting infinite loading when waiting for confirmation for Omni bridge

Regarding Connext claiming - everything works fine now EXCEPT:
- our Connext bridge implementation is incomplete - every bridging that is made through it has expiry date. If user doesn't claim till then bridging goes to 'Cancelled' state on Connext and and hell knows how to get funds back (maybe directly through contract interaction, I haven't looked yet). We don't have any information nor expiration handling which will be frustrating to the users.
image
- as for now I have no good fix for this problem as it would require from me doing some research around Connext ecosystem and figuring out how to handle cancelled bridgings.

File: src/services/EcoBridge/Socket/Socket.selectors.ts [hunk 1]
```diff
@@ -44,6 +44,13 @@ const createSelectPendingTransactions = (selectOwnedTxs: ReturnType<typeof creat
     return pendingTxs
   })
 
+const createSelectFailedTransactions = (selectOwnedTxs: ReturnType<typeof createSelectOwnedTransactions>) =>
+  createSelector(selectOwnedTxs, ownedTxs => {
+    const failedTxs = ownedTxs.filter(tx => tx.status === SocketTxStatus.ERROR && !tx.partnerTxHash)
```
Comment (adamazad) [inline src/services/EcoBridge/Socket/Socket.selectors.ts:49]: I could be wrong, but it should be `ll` instead no?

**Review by adamazad — APPROVED** [review]
Looks good to me!

**Comment by MilanVojnovic95** [comment]
Retested and still I get this 
image

The only thing that has been successfully changed is that the "cancelled" status for successful transactions is no longer displayed now it's "Confirmed" status.
Everything else seems ok. 👍  ✅

**Pull request merged**